server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -391,6 +391,10 @@ public void connectToNode(DiscoveryNode node) throws ConnectTransportException {
         connectToNode(node, (ConnectionProfile) null);
     }
 
+    public void connectToNode(final DiscoveryNode node, boolean fromExtension) {
```

Comment:
Just for my understanding here. Are we creating this new method just for extensions or will it have any impact at other places?

Reply:
Just for extensions - no other impact